packages/router/utils.ts

```diff
@@ -629,27 +631,35 @@ export function generatePath<Path extends string>(
     path = path.replace(/\*$/, "/*") as Path;
   }
 
-  return path
```

Comment:
This multiple `.replace` approach would have potentially exploded out so I went with a `split/map/join` which I think makes it a bit easier to follow 🤷‍♂️